### ISSUE TITLE
User Guide, Dev guide, and User guide standards, should be conformed to Microsoft's style for key names

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#17078]
### Is your feature request related to a problem? Please describe.

In NVDA's documents and interface, key names are written in the style:

```
NVDA+alt+p
```

As @seanbudd pointed out in https://github.com/nvaccess/nvda/issues/16438#issuecomment-2071277348, this is contrary to Microsoft's standard.

### Describe the solution you'd like

Key names should be written as:

```
NVDA+Alt+P
```

Not only does this conform to Microsoft's standard, but it is more clear for reading by low vision users, as was pointed out in the comments of #16438 by @CyrilleB79, @britechguy, and others. Additionally, it may be more clear for certain speech synths, e.g. those that pronounce `a` and `A` differently.

### Describe alternatives you've considered

Leave as is (not desirable, IMO).

### Additional context

In general, I propose adopting the style as described here, in a proposed modification to the user guide standards:

```markdown
- Key commands (e.g. `NVDA+Control+UpArrow`):
  - should have multiple keys in a single gesture separated with plus signs (+)
  - should be written in Upper Camel Case, with no spaces; e.g. `Numpad7`, `NVDA+Alt+P`
  - single letters should be capitalized, as described in [Microsoft's Key Names](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/keys-keyboard-shortcuts#key-names)
  - should be encapsulated in monospace code-block formatting
  - NVDA should be capitalized
```

While Upper Camel Case isn't strictly necessary here, it seems easier to write that, than to explain that we want Lower Camel Case, except that single words such as Shift or Control, when not combined with another word, should be capitalized. There is no name for that. It seems more clear just to switch to Upper Camel Case.

An alternative would be to switch to Microsoft's "Sentence Capitalization" as they call it in that article. But that puts spaces between each word, and lower cases second words, and I don't think we would find those desirable given the way we write key names.

**Comment by CyrilleB79** [comment]
As for any reformatting proposal, as discussed in #16438, I'd wonder what would be the best solution / process to avoid translators to have to re-format their translated strings.

E.g. if you reformat the key gestures in the English documentation for NVDA 2025.1, translators will end up with many modified lines. But even if they use an automatic regexp replacement in the `.po` file, they cannot know if the change is only due to reformatting or if there are other changes. Thus they will have to review all these strings in any case.

Also, re the adopted format, it is worth noting that camel case convention applies to latin alphabet, but not to all types of writing as written by @josephsl in  https://github.com/nvaccess/nvda/issues/16438#issuecomment-2069543390. I do not know what the impact is; maybe not so much since we already use camel case.

One other point is that other docs in the repo should be scanned and key names converted. As a first step, we may look for the regexp `[A-Za-z]+\+[A-Za-z]+` in all the `.md` files in the repo; it's not exhaustive but it can give an idea where such keys are defined.

**Comment by XLTechie** [comment]
Hmm. Given that there is not a line-to-line correspondence between the English user guide, and the guides of other languages, this would be hard to automate indeed.

It seemed, from my reading of @michaelDCurran's plan for migrating user docs to Crowdin, there may start to be a line-to-line correspondence in the future. But perhaps that was a misunderstanding on my part.

**Comment by CyrilleB79** [comment]
> Hmm. Given that there is not a line-to-line correspondence between the English user guide, and the guides of other languages, this would be hard to automate indeed. It seemed, from my reading of @michaelDCurran's plan for migrating user docs to Crowdin, there may start to be a line-to-line correspondence in the future. But perhaps that was a misunderstanding on my part.

There's a push, and I am partly responsible for it, to encourage translators to have a line to line correspondance in the User Guide and the change log. In the SVN repo where the translation of the documentation still takes place (but probably not for long), we have automatic script to compare the structure of the documents between English and translated content and report differences. This script not only guarantees that the number of line is the same, but also checks structural pattern for each matching line such as title formatting, level of title, anchors, links to anchors, number of backquotes, initiating with bullet, etc. The script issues a report listing all the differences found, or no difference if the language is structurally up-to-date.

According to Sascha in message [#4218](https://groups.io/g/nvda-translations/message/4218) on the translators mailing list:
> Note that 18 languages for the user guide currently match completely: bg, da, de, es, fa, fi, fr, hr, it, pl, ru, sk, sr, tr, uk, vi, zh_CN and zh_TW.

And some of them are not far from being structurally identical (pt_PT). The majority of the remaining languages are not up-to-date, sometimes a lot, so they will have to review many strings in any case to catch up.

On NVDA's GitHub, they probably do not match because we have just started the 2024.4 beta phase.

If you look at the messages on the [translators mailing list](https://groups.io/g/nvda-translations/topics), you may have more context.

After the transition to crowdin, I do not know if and how we will be able to check the structure content. Some part of it will be not even need to be checked (titles, level, bullets) since they will not be translated anymore. But for anchors or code formatting with backquotes, it's less clear.

Cc @michaelDCurran in case this work should be done just before the switch to Crowdin (may be late) or just after, at least for fully synchronized languages.

**Comment by michaelDCurran** [comment]
My thoughts on this:
* Definitely not before we complete the move to Crowdin. The user guide 
is already being uploaded and tracked on Crowdin, though translation has 
not officially started yet. Changing a significant amount of lines at 
this stage would be very confusing to the whole project.
* The new system ensures  line-for-line, and as prefix / suffix 
structure on the line is not translatable, the general structure of the 
document is maintained.
* Syntax within the translated content such as inline code, links etc, 
is currently not validated on Crowdin, but I believe it is possible to 
write translation rules in Crowdin which can handle this. I'll look into 
this after we get the main part done.
* Even when the move to Crowdin is complete, It will be quite hard to 
handle the significant rewrite of gestures, without causing massive work 
for translators. As much as conforming to a standard is nice, NVDA's 
current gesture style is not ambiguous and therefore I really don't 
think the advantages of changing outweigh the disadvantages.

**Comment by Adriani90** [comment]
I agree with Mick fully. In my view the effort and the impact are not balanced. The workload would be huge for a minimalistic user benefit. People with a bit of vision ability have other possibilities to zoom the text, capital letters will not really add such a huge improvement to accessibility.
As I said also in another context regarding the user guide, adapting it to Microsoft standards won‘t make it much better. But this is just my opinion.

**Comment by CyrilleB79** [comment]
Except if you find a way and want to modify all the existing content (all .md in the repo, including translated content), I'd share the same opinion as @michaelDCurran and @Adriani90.

I.e.:
* MS style would be nicer than NVDA's current style
* Today, NVDA style is used quite exclusively in all the content; this consistency is a real plus
* A mix between convensions (MS style vs. current NVDA style) would be the worse option, leading to more confusion.
* Modifying only the English content and letting translators propagate the change is not an option since:
  * it's a lot of work for them with little benefit
  * the experience shows that a significant part of the translated content will remain with the old style, breaking the current consistency in gesture formatting

**Comment by XLTechie** [comment]
While I think there may be some possibility of automating this change across the translation base, that can't be investigated until Crowdin is fully in use, and the one-to-one line alignment is attained.
This might as well be closed until then.
